lib/std/fs/Dir.zig

```diff
@@ -11,14 +11,23 @@ pub const Entry = struct {
 
 const IteratorError = error{ AccessDenied, SystemResources } || posix.UnexpectedError;
 
+pub const IterateOptions = struct {
+    /// If it's known that the `Dir` has not had its cursor modified yet (e.g. it was just opened),
+    /// this can be set to false to skip an explicit reset of the cursor.
+    reset_cursor: bool = true,
+
+    /// Explicitly stat the files, if the file kind is not reported by the underlying directory reading operation.
+    query_kind: bool = true,
```

Comment:
I think this would benefit from being more detailed, and potentially with a name change of the field:

```suggestion
    /// Some operating systems may return 'unknown' as the entry kind when the underlying 
    /// filesystem does not support reporting the entry kind during directory iteration. If this
    /// field is set to true, then an additional `stat` call is performed in order to attempt to resolve
    /// the `kind` whenever the kind is initially returned as 'unknown'.
    stat_on_unknown_kind: bool = true,
```

Reply:
That doesn't quite catch it (too specific for Linux). Some Operating Systems (like Solaris) just outright don't support getting the file types with the directory iteration.

My idea for a better description:
 ```zig
/// Query the kind of the iterated files
///
/// Some operating systems or filesystems don't provide the kind of the entries being iterated.
/// In these cases an additional `stat` call is necessary to determine the kind.
/// If the kind is not needed or the stat call is performed manually (e.g. to also get additional `stat` data), this option can be disabled.
/// In this case the `kind` field will contain `unknown`, when the file type is not provided by the directory iteration call.
query_kind: bool = true,
```